pkg/translator/sumoicmarshaler/sumo_ic_marshaler.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type SumoICMarshaler struct {
+	format string
+}
+
+func (marshaler *SumoICMarshaler) Format() string {
+	return marshaler.format
+}
+
+func NewSumoICMarshaler() SumoICMarshaler {
+	return SumoICMarshaler{}
+}
+
+func logEntry(buf *bytes.Buffer, format string, a ...interface{}) {
+	buf.WriteString(fmt.Sprintf(format, a...))
+	buf.WriteByte('\n')
+}
+
+const (
+	SourceCategoryKey = "_sourceCategory"
+	SourceHostKey     = "_sourceHost"
+	SourceNameKey     = "log.file.path_resolved"
+)
+
+func (SumoICMarshaler) MarshalLogs(ld plog.Logs) ([]byte, error) {
+	buf := bytes.Buffer{}
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		sourceCategory, exists := rl.Resource().Attributes().Get(SourceCategoryKey)
+		if exists == false {
+			return nil, errors.New("_sourceCategory attribute does not exists")
+		}
+		sourceHost, exists := rl.Resource().Attributes().Get(SourceHostKey)
+		if exists == false {
+			return nil, errors.New("_sourceHost attribute does not exists")
+		}
+		ills := rl.ScopeLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			logs := ils.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				lr := logs.At(k)
+				dateVal := lr.ObservedTimestamp()
+				body := lr.Body().AsString()
+				sourceName, exists := lr.Attributes().Get(SourceNameKey)
+				if exists == false {
+					return nil, errors.New("_sourceName attribute does not exists")
+				}
+				logEntry(&buf, "{\"data\": \"%s\",\"sourceName\":\"%s\",\"sourceHost\":\"%s\",\"sourceCategory\":\"%s\",\"fields\":{},\"message\":\"%s\"}",
+					dateVal, sourceName.AsString(), sourceHost.AsString(), sourceCategory.AsString(), body)
```

Comment:
Yes, this is marshalling to json. We can follow your suggestion and marshal a struct. I just wanted to avoid this additional transformation, however it might be cleaner. Regarding `fields` I was not sure what be there, so for now it's a placeholder